packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -143,28 +520,50 @@ class HttpGatewayRouteSpec extends GatewayRouteSpec {
    */
   readonly routeType: Protocol;
 
+  /**
+   * The gateway route action to rewrite.
+   *
+   * @default - rewritten to the target Virtual Service's hostname and the matched prefix or path is rewritten to '/'.
+   */
+  readonly rewrite?: HttpGatewayRouteRewrite;
+
   constructor(options: HttpGatewayRouteSpecOptions, protocol: Protocol.HTTP | Protocol.HTTP2) {
     super();
     this.routeTarget = options.routeTarget;
     this.routeType = protocol;
     this.match = options.match;
+    this.rewrite = options.rewrite;
   }
 
-  public bind(_scope: Construct): GatewayRouteSpecConfig {
-    const prefixPath = this.match ? this.match.prefixPath : '/';
-    if (prefixPath[0] != '/') {
-      throw new Error(`Prefix Path must start with \'/\', got: ${prefixPath}`);
+  public bind(scope: Construct): GatewayRouteSpecConfig {
+    const prefix = !isMatchPropertiesUndefined(this.match) ? this.match?.pathOrPrefix?.bind(scope).requestMatch.prefix :'/';
+
+    if (prefix && prefix[0] != '/') {
+      throw new Error(`Prefix Path must start with \'/\', got: ${prefix}`);
+    }
+    if (this.rewrite?.pathOrPrefix?.bind(scope).prefix && !prefix) {
+      throw new Error('HTTP Gateway Route Prefix Match must be set.');
     }
+    if (this.rewrite?.pathOrPrefix?.bind(scope).path && prefix) {
+      throw new Error('HTTP Gateway Route Prefix Match and Path Rewrite both cannot be set.');
+    }
+
     const httpConfig: CfnGatewayRoute.HttpGatewayRouteProperty = {
       match: {
-        prefix: prefixPath,
+        prefix: prefix,
+        path: this.match?.pathOrPrefix?.bind(scope).requestMatch.path,
+        hostname: this.match?.hostname?.bind(scope).hostnameMatch,
+        method: this.match?.method,
+        headers: this.match?.headers?.map(header => header.bind(scope).httpRouteHeader),
+        queryParameters: this.match?.queryParameters?.map(queryParameter => queryParameter.bind(scope).queryParameter),
       },
       action: {
         target: {
           virtualService: {
             virtualServiceName: this.routeTarget.virtualServiceName,
           },
         },
+        rewrite: renderHttpGatewayRouteRewrite(scope, this.rewrite),
```

Comment:
Since value returned by `bind()` method may not be idempotent, should this render method accept each variables (`prefixRewrite` and `pathRewrite`) rather than just accepting `this.rewrite`?

Reply:
I think `renderHttpGatewayRouteRewrite()` should take as an argument not a `HttpGatewayRouteRewrite`, but a `HttpGatewayRouteRewriteConfig`.

Reply:
Well noted. Making this change.